js/ui/grid_core/ui.grid_core.grid_view.js

```diff
@@ -131,13 +131,25 @@ const ResizingController = modules.ViewController.inherit({
     },
 
     _getBestFitWidths: function() {
+        const rowsView = this._rowsView;
+        const columnHeadersView = this._columnHeadersView;
+
         if(!this.option('legacyRendering')) {
-            return this._rowsView.getColumnWidths();
+            let widths = rowsView.getColumnWidths();
+
+            if(!widths?.length) {
+                const headersTableElement = columnHeadersView._tableElement;
```

Comment:
Maybe use _getTableElement and _setTableElement  methods? Can you make them public?